src/System.Net.Requests/src/System/Net/RequestStream.cs

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace System.Net
+{
+    // Cache the request stream into a MemoryStream.  This is the
+    // default behavior of Desktop HttpWebRequest.AllowWriteStreamBuffering (true).
+    // Unfortunately, this property is not exposed in .NET Core, so it can't be changed
+    // This will result in inefficient memory usage when sending (POST'ing) large
+    // amounts of data to the server such as from a file stream.
+    internal class RequestStream : Stream
+    {
+        private MemoryStream _buffer = new MemoryStream();
+
+        public RequestStream()
+        {
+        }
+
+        public override bool CanRead
+        {
+            get
+            {
+                return false;
+            }
+        }
+
+        public override bool CanSeek
+        {
+            get
+            {
+                return false;
+            }
+        }
+
+        public override bool CanWrite
+        {
+            get
+            {
+                return true;
+            }
+        }
+
+        public override void Flush()
+        {
+            // Nothing to do.
+        }
+
+        public override long Length
+        {
+            get
+            {
+                throw new NotSupportedException();
+            }
+        }
+
+        public override long Position
+        {
+            get
+            {
+                throw new NotSupportedException();
+            }
+            set
+            {
+                throw new NotSupportedException();
+            }
+        }
+
+        public override int Read(byte[] buffer, int offset, int count)
+        {
+            throw new NotSupportedException();
+        }
+
+        public override long Seek(long offset, SeekOrigin origin)
+        {
+            throw new NotSupportedException();
+        }
+
+        public override void SetLength(long value)
+        {
+            throw new NotSupportedException();
+        }
+
+        public override void Write(byte[] buffer, int offset, int count)
+        {
+            _buffer.Write(buffer, offset, count);
+        }
```

Comment:
Consider overriding both WriteAsync and FlushAsync to also just delegate to _buffer.WriteAsync and _buffer.FlushAsync.  As it's a memory stream, it'll perform the operations synchronously, whereas if they remain not overridden, it'll default to the base stream's implementation which will queue work items that just call Write and Flush, respectively.